### ISSUE TITLE
Resource must be configured with notify=true

### ISSUE THREAD

**Issue by manchuwook** [MicrosoftDocs/sql-docs#1841]
When I'm using this guide, I get to the part where I create the resource and now it just has a bunch of FAILED (blocked) entries in pcs status.

````
Cluster name:SubCluster
Stack: corosync
Current DC: server2.sub.mydomain.com (version 1.1.18-2b07d5c5a9) - partition with quorum
Last updated: Mon Apr  1 00:53:26 2019
Last change: Mon Apr  1 00:45:15 2019 by root via cibadmin on server0.sub.mydomain.com

4 nodes configured
1 resource configured

Online: [ server0.sub.mydomain.com server1.sub.mydomain.com server2.sub.mydomain.com server3.sub.mydomain.com ]

Full list of resources:

SubHAGC     (ocf::mssql:ag):        FAILED (blocked)[ server3.sub.mydomain.com server0.sub.mydomain.com server1.sub.mydomain.com server2.sub.mydomain.com ]

Failed Actions:
*SubHAGC_stop_0 on server3.sub.mydomain.com 'not configured' (6): call=41, status=complete, exitreason='Resource must be configured with notify=true',
    last-rc-change='Mon Apr  1 00:53:06 2019', queued=0ms, exec=32ms
*SubHAGC_stop_0 on server0.sub.mydomain.com 'not configured' (6): call=41, status=complete, exitreason='Resource must be configured with notify=true',
    last-rc-change='Mon Apr  1 00:53:06 2019', queued=0ms, exec=34ms
*SubHAGC_stop_0 on server1.sub.mydomain.com 'not configured' (6): call=41, status=complete, exitreason='Resource must be configured with notify=true',
    last-rc-change='Mon Apr  1 00:53:06 2019', queued=0ms, exec=89ms
*SubHAGC_stop_0 on server2.sub.mydomain.com 'not configured' (6): call=41, status=complete, exitreason='Resource must be configured with notify=true',
    last-rc-change='Mon Apr  1 00:53:06 2019', queued=0ms, exec=169ms


Daemon Status:
  corosync: active/enabled
  pacemaker: active/enabled
  pcsd: active/enabled
````

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: f7de1f2c-d879-9ce4-09fd-251fc3dacc05
* Version Independent ID: 4de58f31-0b21-f4da-2a43-1eb4e8151f00
* Content: [Configure Ubuntu Cluster for SQL Server Availability Group - SQL Server](https://docs.microsoft.com/en-us/sql/linux/sql-server-linux-availability-group-cluster-ubuntu?view=sql-server-2017)
* Content Source: [docs/linux/sql-server-linux-availability-group-cluster-ubuntu.md](https://github.com/MicrosoftDocs/sql-docs/blob/live/docs/linux/sql-server-linux-availability-group-cluster-ubuntu.md)
* Product: **sql**
* Technology: **linux**
* GitHub Login: @MikeRayMSFT
* Microsoft Alias: **mikeray**

**Comment by manchuwook** [comment]
This is on an Ubuntu cluster

**Comment by MightyPen** [comment]
@MikeRayMSFT Hi Mike, Here is an Issue 1841 about one of your SQL Linux articles.    I am unsure whether the original poster needs to provide more info, I leave that up to you.

Thanks.

**Comment by manchuwook** [comment]
I tried both
````
sudo pcs resource create SubCluster ocf:mssql:ag ag_name=SubHAG meta failure-timeout=30s --master meta notify=true
````

and 

````
sudo pcs resource create SubCluster ocf:mssql:ag ag_name=SubHAG meta failure-timeout=30s meta notify=true --master 
````

as I had seen something about a bug with putting meta notify=true after --master and they both have the same resulting error message.

**Comment by MikeRayMSFT** [comment]
@manchuwook 
Thank you for submission.
I've created a work item to track this issue (internal tracking number 1515521).

I'm checking internally to see your best course of action to get some assistance.

**Comment by MikeRayMSFT** [comment]
The error message reported indicates the exit reason as “Resource must be configured with notify=true”

We cannot tell for sure as to what command was used to create the resource. As per the command in the document, it seems like customer did not specify this 

`sudo pcs resource create ag_cluster ocf:mssql:ag ag_name=ag1 meta failure-timeout=30s --master meta notify=true`

**Comment by vnextcoder** [comment]
I was working on Ubuntu to create a SQL 2019 cluster, however, it appears that the Installation itself was faulty.  In my case, the ag_metadata xml itself didn't contain any "notify" parameter. 

I also tried updating the /usr/lib/ocf/lib/mssql/ag_metadata file with "notify" parameter, but still the "ag" script (part of mssql-server-ha) and which is also used by the PaceMaker /Corosync to connect to SQL Nodes, it still didn't work. 

Since it was just a learning and doing POC, I gave up after 1 full day of hard work to get things working. I would retry few weeks later to check if new versions actually fix the issue.
